cadc-access-control/src/main/java/ca/nrc/cadc/ac/client/GMSClient.java

```diff
@@ -151,51 +153,84 @@ public String getEventHeader()
     {
         return null; // no custom eventID header
     }
-    
 
-    
     /**
      * GMSClient Interface compliance.
-     * 
+     *
      * Default 'role' within a group is 'membership'
-     * 
+     *
      * Ensure serviceIDs match.
      */
     @Override
-    public boolean isMember(GroupURI group) {
-        if (group == null) {
-            throw new IllegalArgumentException("Null group");
+    public List<GroupURI> getMemberships(List<GroupURI> groups) {
+        // return empty list for null or empty groups list
+        if (groups == null || groups.isEmpty()) {
+            return new ArrayList<>();
         }
-        if (!group.getServiceID().equals(serviceID)) {
-            throw new UnsupportedOperationException("Group is not in the target GMS service.");
+
+        // discard groups not in the target GMS service
+        List<String> groupNames = new ArrayList<>();
+        for (GroupURI group : groups) {
+            if (group.getServiceID().equals(this.serviceID)) {
+                groupNames.add(group.getName());
+            } else {
+                log.warn(String.format("%s is not in the target GMS service %s",
```

Comment:
Since we decided "ignore" was the right policy, this should be logged as debug.

Reply:
Changed from warn to debug.